### PR TITLE
ZFIN-8985 Replace criteria objects

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

File: source/org/zfin/expression/repository/HibernateExpressionRepository.java [hunk 1]
```diff
@@ -1390,34 +1380,43 @@ public boolean pileStructureExists(ExpressedTermDTO expressedTerm, String public
...
-        publication.add(Restrictions.eq("zdbID", publicationID));
-        Criteria superterm = crit.createCriteria("superterm");
-        superterm.add(Restrictions.eq("termName", supertermName));
-
-        List list = crit.list();
-        return list != null && !list.isEmpty();
+        hql += " where " + String.join(" and ", hqlClauses);
+        Query<ExpressionStructure> query = session.createQuery(hql, ExpressionStructure.class);
+        parameterMap.forEach(query::setParameter);
+        return query.list() != null && !query.list().isEmpty();
```
Comment (rtaylorzfin) [inline source/org/zfin/expression/repository/HibernateExpressionRepository.java:1419]: Does this run the query twice? (once for `query.list() != null` and once for `!query.list().isEmpty()`?)

File: source/org/zfin/profile/repository/HibernateProfileRepository.java [hunk 2]
```diff
@@ -38,1007 +35,1038 @@
...
+        if (curator == null) {
+            return null;
+        }
+
+        Session session = currentSession();
+        String hql = """
+            from CuratorSession
+            where curator.zdbID = :curatorID
+            AND publication.zdbID = :pubID
+            AND field = "field
```
Comment (rtaylorzfin) [inline source/org/zfin/profile/repository/HibernateProfileRepository.java:125]: `"field` should be `:field`

File: source/org/zfin/sequence/repository/HibernateSequenceRepository.java [hunk 3]
```diff
@@ -326,119 +328,83 @@ public List<DBLink> getDBLinks(String accessionString, ReferenceDatabase... refe
         if (accessionString.contains("."))
             truncatedAccession = accessionString.substring(0, accessionString.indexOf("."));
         Session session = HibernateUtil.currentSession();
-        Criteria criteria = session.createCriteria(DBLink.class);
-        Criterion c1 = Restrictions.eq("accessionNumber", accessionString);
-        Criterion c2 = Restrictions.eq("accessionNumber", truncatedAccession);
-        criteria.add(Restrictions.or(c1, c2));
+        String hql = "from DBLink where " +
+                     " (accessionNumber = :accessionNumber1 or accessionNumber = :accessionNumber2) order by referenceDatabase, accessionNumber";
```
Comment (rtaylorzfin) [inline source/org/zfin/sequence/repository/HibernateSequenceRepository.java:332]: this `order by ...` should be removed

File: source/org/zfin/expression/repository/HibernateExpressionRepository.java [hunk 4]
```diff
@@ -2108,111 +2107,67 @@ public List<Figure> getFigures(ExpressionSummaryCriteria expressionCriteria) {
...
-        if (expressionCriteria.isChemicalEnvironment()) {
-            aliasMap.put("expressionExperiment", "xpatex");
-            aliasMap.put("xpatex.fishExperiment", "genox");
-            aliasMap.put("genox.experiment", "exp");
-            aliasMap.put("exp.experimentConditions", "cond");
-            aliasMap.put("cond.conditionDataType", "cdt");
-            criteria.add(Restrictions.eq("cdt.group", "chemical"));
-            logger.debug("chemical environments only");
+            hqlClauses.add("expressionFigureStage.expressionExperiment.fishExperiment.standardOrGenericControl = true");
         }
```
[the remark this answers is not in the document]
Comment (cmpich) [reply source/org/zfin/expression/repository/HibernateExpressionRepository.java:2157]: Yes, that was not used any longer.

**Review by rtaylorzfin — APPROVED** [review]
Nice work!

**Pull request merged**